sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -175,6 +175,30 @@ namespace Azure { namespace Core { namespace Http {
     NetworkAuthenticationRequired = 511, ///< HTTP 511 Network Authentication Required.
   };
 
+  /**
+   * @brief Defines a range of bytes within an HTTP resource, starting at an `Offset` and ending at
+   * `Offset + Length - 1` inclusively.
+   *
+   */
+  struct Range
+  {
+    /**
+     * @brief The starting point of the HTTP Range.
+     *
+     */
+    int64_t Offset = 0;
+
+    /**
+     * @brief The size of the HTTP Range.
+     *
+     */
+    int64_t Length = 0;
```

Comment:
Make `Length` nullable.

Reply:
FWIW, I prefer this to my suggestions regarding the sentinel value.

Reply:
+1 for nullable